copy_methods.go

```diff
@@ -0,0 +1,62 @@
+package copy
+
+import (
+	"errors"
+	"io"
+	"os"
+)
+
+// ErrUnsupportedCopyMethod is returned when the FileCopyMethod specified in
+// Options is not supported.
+var ErrUnsupportedCopyMethod = errors.New(
```

Comment:
What's the case when we use it?

Reply:
So the user can tell if the copy failed because of no support, instead of another error like permission:

```go
err := copy.Copy("/from", "/to", copy.Options {
    FileCopyMethod: copy.ReflinkCopy,
})

if errors.Is(err, copy.ErrUnsupportedCopyMethod) {
    // Retry with other method
    err = copy.Copy("/from", "/to", copy.Options {
        FileCopyMethod: copy.CopyBytes,
    })
}
```

Reply:
quick question: Do we use this error in this current version?

Reply:
Not yet. It's used in a later pull request I made

Reply:
OK, got it. In this case I'll merge it.

Very personally speaking, I would not like to mix what we don't use at that moment. Similar idea to YAGNI.

Reply:
I can remove it and add it back in the other PR, if that's preferred?

Reply:
No need. Thanks for your consideration. I understand why it's here and we don't have any doubt we will need it ;)